ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala

```diff
@@ -100,32 +105,81 @@ class TransactionCommitterSpec extends WordSpec with Matchers with MockitoSugar
     }
   }
 
-  "validateLedgerTime" should {
-    "continue without accessing ledger configuration if record time is not available" in {
-      val instance = createTransactionCommitter()
-      val context = new FakeCommitContext(recordTime = None)
-      val actual = instance.validateLedgerTime(context, aTransactionEntrySummary)
+  "validateLedgerTime" can {
+    "when the record time is not available" should {
+      val configurationStateValue = defaultConfigurationStateValueBuilder.build
+      val inputWithDeclaredConfig =
+        Map(Conversions.configurationStateKey -> Some(configurationStateValue))
+      val context = new FakeCommitContext(recordTime = None, inputs = inputWithDeclaredConfig)
 
-      actual match {
-        case StepContinue(_) => succeed
-        case StepStop(_) => fail
+      "continue" in {
+        val result = instance.validateLedgerTime(context, aTransactionEntrySummary)
+
+        result match {
+          case StepContinue(_) => succeed
+          case StepStop(_) => fail
+        }
+      }
+
+      val aSubmissionTime = createProtobufTimestamp(seconds = 1)
```

Comment:
It's personal taste but I find test cases highly unreadable if I have to read through lots of boilerplate first before I get to the actual test case definition. I'd suggest either keeping these near the top where they were or making them `private def`s and putting them at the end of the file.

Reply:
Yes I agree, moved everything at the top.